bee-network/src/tests/common/keys_and_ids.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::PeerId;
+
+pub fn gen_constant_peer_id() -> PeerId {
+    "12D3KooWJWEKvSFbben74C7H4YtKjhPMTDxd7gP7zxWSUEeF27st".parse().unwrap()
+}
+
+#[cfg(feature = "full")]
+mod __full {
+
+    use super::*;
+    use libp2p::identity::{
+        ed25519::{Keypair, SecretKey},
+        PublicKey,
+    };
+    use std::iter::repeat;
+
+    pub fn gen_random_peer_id() -> PeerId {
+        PeerId::from_public_key(libp2p_core::PublicKey::Ed25519(Keypair::generate().public()))
+    }
+
+    pub fn gen_deterministic_peer_id(gen: impl ToString) -> PeerId {
+        let keys = gen_deterministic_keys(gen);
+        PeerId::from_public_key(PublicKey::Ed25519(keys.public()))
+    }
+
+    pub fn gen_deterministic_keys(gen: impl ToString) -> Keypair {
+        let hex_chars = ('0'..'9').chain('a'..'f').collect::<Vec<_>>();
```

Comment:
This is just https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_hexdigit, no? Or are you specifically avoiding uppercase letters?

Reply:
damn, I should have read this comment first, and not doing them in reverse :joy:

Reply:
I think I coded this at 4am in the morning or so. There were no brain cells left to think of `is_ascii_hexdigit`. :joy:

Reply:
`char` has some lovely methods. Insane that a lot of other languages don't have simple stuff like this!